leader-election-sf/WorkerStatelessService/WorkerStatelessService.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Fabric;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ServiceFabric.Services.Runtime;
+using Shared.Service;
+using Microsoft.ServiceFabric.Services.Remoting.Client;
+using Microsoft.ServiceFabric.Services.Client;
+
+namespace WorkerStatelessService
+{
+    /// <summary>
+    /// An instance of this class is created for each service instance by the Service Fabric runtime.
+    /// </summary>
+    internal sealed class WorkerStatelessService : StatelessService
+    {
+        public WorkerStatelessService(StatelessServiceContext context)
+            : base(context)
+        { }
+
+        /// <summary>
+        /// This is the main entry point for your service instance.
+        /// </summary>
+        /// <param name="cancellationToken">Canceled when Service Fabric needs to shut down this service instance.</param>
+        protected override async Task RunAsync(CancellationToken cancellationToken)
+        {
+            // TODO: Replace the following sample code with your own logic 
+            //       or remove this RunAsync override if it's not needed in your service.
+
+            // NOTE: Waits to "ensure" Leader starting first. Not realy needed.
+            //       Not sure if there's a better way (i.e. service dependency enforcement into SF).
+            //       Altenatives in SO answer here:
+            //       http://stackoverflow.com/questions/41370742/azure-service-fabric-specify-service-application-startup-dependency
+            await Task.Delay(20000, cancellationToken);
+
```

Comment:
@woodp This seems to be a convoluted approach i.e. waiting and hoping that the service on which my service is dependent will boot up within a given interval. We (p&p) usually do not publish something that we're not confident about. I'd recommend investigating other approaches such as probing the leader service using a REST call as per the spec for [GetServiceInfo](https://docs.microsoft.com/en-us/rest/api/servicefabric/sfclient-api-getserviceinfo). And it seems that HTTP 200 in this case is good enough to ensure that service is up and running. I'm suspending further review till we've a convincing resolution for this.